opencog/atoms/base/Atom.h

```diff
@@ -173,6 +173,11 @@ class Atom
 
     virtual ~Atom();
 
+    virtual const std::string& getName() const {
+        static std::string empty_string("");
+        return empty_string;
```

Comment:
I can add that, but the C++ code was previously doing instance checking. i.e. NodeCast and then checking for null returns.